bosh-director/lib/bosh/director/instance_updater/vm_updater.rb

```diff
@@ -55,8 +55,6 @@ def detach
 
       vm_deleter = VmDeleter.new(@instance, @vm_model, @cloud, @logger)
       vm_deleter.delete
-
-      @instance.job.resource_pool.add_idle_vm
```

Comment:
It was mentioned in the description of task: "bosh stop --hard is useful for releasing all of the compute resources without losing any persistent data". Idle vm keeps resources to use.
Additionally, the task was "bosh stop --hard job1 should no longer show that vm in bosh vms"  - this idle vm is showed in bosh vms.